Sources/AsyncStream.swift

```diff
@@ -1,18 +1,24 @@
 public protocol AsyncSending {
-    func send(_ data: Data, timingOut deadline: Double, completion: (Void throws -> Void) -> Void)
+    associatedtype Sent
+    func send(_ data: Sent, timingOut deadline: Double, completion: (Void throws -> Void) -> Void)
```

Comment:
Can we typealias the completion function as something else, these throwable closure as first arg of closure still look funny. We can also comment on the typealias as to usage.